packages/internals/src/wasm.ts

```diff
@@ -0,0 +1,20 @@
+import _prismaFmt from '@prisma/prisma-fmt-wasm'
+import { match } from 'ts-pattern'
+
+import { getWASMVersion } from './engine-commands/getEngineVersion'
+import { BinaryType } from './resolveBinary'
+
+/**
+ * Re-exports Prisma WASM modules with an overridden `version()` method that returns the npm/hash version of the given WASM engine.
+ */
+
+export const prismaFmt = new Proxy(_prismaFmt, {
+  get(target, prop) {
+    return match(prop)
+      .with('version', () => () => {
+        const overriddenVersion = getWASMVersion(BinaryType.prismaFmt)
+        return overriddenVersion
+      })
+      .otherwise(() => target[prop])
+  },
+})
```

Comment:
Note: this is the reason why the CI is slightly slower when performing `prisma version`, but we argue [here](https://prisma-company.slack.com/archives/C02A1646TBR/p1660658408938849?thread_ts=1660638787.096929&cid=C02A1646TBR) that it's fine actually.